src/error.rs

```diff
@@ -0,0 +1,7 @@
+use thiserror::Error;
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum Error {
+    #[error("{0} is not a valid absolute IRI")]
```

Comment:
You haven't implemented IRIs here (purely by virtue of `String` not supporting `utf-16`).
```suggestion
    #[error("{0} is not a valid absolute URI")]
```